content/blog/2021/04/2021-04-16-deprecated-ruby-runtime.adoc

```diff
@@ -0,0 +1,129 @@
+plugins to be written in ruby but still get integrated into the main
+java Jenkins process.
+
+Over the years though, the experiment has stopped being maintained, the
+dozen or so plugins that use it take a lot of user effort to make work,
+and the Jenkins ruby runtime does not support Java 11.
+Specifically the last few years have been really hard on the Jenkins
+core developers, as they still have to maintain all the hacks and
+processes to keep the ruby runtime barely working.
+
+One month from now, May 15th, 2021, the Jenkins update center will stop distributing
+ruby-runtime and its https://plugins.jenkins.io/ruby-runtime/#dependencies[associated plugins].
+
+== What does this mean for you?
+
+If you are one of the few users who are using the following plugins. The
+only change will come for new installs. Suspended plugins can stay
+installed, but can not be newly installed without manually downloading
+releases.
+
+The lists below provide additional information about the most popular plugins based on the Ruby runtime.
+Please plan your transition away from these plugins.
+They will be removed from the update center May 15, 2021.
+
+=== Most popular Ruby runtime plugins
+
+plugin:gitlab-hook[Gitlab Hook]::
+Last released **5 years ago**.
+Contains multiple security vulenerabilties.
+Suggestion: Use the plugin:gitlab-plugin[GitLab] plugin and the plugin:gitlab-branch-source[GitLab Branch Source Plugin].
+
+plugin:cucumber[Cucumber]::
+Last released **8 years ago**.
+Suggestion: Use `sh` or `bat` to run `cucumber` from the command line.
+
+plugin:pyenv[pyenv]::
+Last released **7 years ago**.
+Suggestion: Use `sh` or `bat` to run `pyenv` from the command line.
+
+plugin:rvm[Rvm]::
+Last released **5 years ago**.
+Suggestion: Use `sh` or `bat` to run `rvm` from the command line.
+
+=== Fewer than 1000 installations
+
+plugin:capitomcat[Capitomcat]::
+Last released **6 years ago**.
+
+Suggestion: Install Ruby and Capistrano and use `sh` or `bat` to invoke them from the command line.
+
+plugin:commit-message-trigger-plugin[Commit Message Trigger]::
+Last released **7 years ago**.
+Suggestion: Use `sh`, `bat`, or other scripts to read git commit messages and conditionally execute Pipeline steps.
+
+plugin:git-notes[git-notes]::
+Last released **9 years ago**.
+Suggestion: Use `sh`, `bat`, or other scripts to run `git` to annotate commits.
+
+plugin:rbenv[rbenv]::
+Last released **5 years ago**.
+Suggestion: Use `sh` or `bat` to run `pyenv` from the command line.
+
+=== Fewer than 500 installations
+
+plugin:chef[Chef]::
+Last released **6 years ago**.
+Suggestion: Use `sh` or `bat` to run `chef` from the command line.
+
+plugin:ci-skip[Ci Skip]::
+Last released **7 years ago**.
+Suggestion: Use `sh`, `bat`, or other scripts to read git commit messages and conditionally execute Pipeline steps.
```

Comment:
IIRC there are some maintained Jenkins plugins to do something similar.

Reply:
https://github.com/jenkinsci/scm-trait-commit-skip-plugin

last commit:
![image](https://user-images.githubusercontent.com/21194782/115037709-a9896880-9ec6-11eb-954f-3cbdce653b1d.png)